server/controllers/medical/patient.js

```diff
@@ -429,3 +452,148 @@ function logVisit(patientData, userId) {
 function isEmpty(array) {
   return array.length === 0;
 }
+
+/**
+* GET /patient/search?name="..."&details=...&limit=...
```

Comment:
These are really good comments.  For query strings, it might be good to specify their type in the example URL.  Like this:
```js
/** GET /patient/search?name={String}&details={Number}&limit={Number} */
```
This was a user can easily see what parameters the query string might take.